.release-notes/4173.md

```diff
@@ -0,0 +1,18 @@
+## Fix for segfault when using Array.copy_to() with an empty Array
+
+Currently, the `Array.copy_to()` function does not check whether the Array is empty or has been instanced with any values. This leads to a runtime crash when calling `copy_to()` on an empty array such as in this example (issue #4172):
```

Comment:
```suggestion
Previously, `Array.copy_to()` did not check whether the source Array had been initialized. Uninitialized arrays internally have a null pointer for their data as no memory has been allocated for them yet. When doing a copy like in the example below, this would result in an attempt to dereference the null pointer and a segfault.
```